src/app/component/key/typer-interface.js

```diff
@@ -0,0 +1,98 @@
+class TyperInterface {
+  constructor() {
+    this.target = null;
+    this.selection = {
+      start: 0,
+      end: 0,
+      get length() {
+        return this.end - this.start;
+      },
+    };
+  }
+
+  getSelection() {
+    if (this.target.selectionStart || this.target.selectionStart === '0') {
+      this.selection.start = this.target.selectionStart;
+      this.selection.end = this.target.selectionEnd;
+    } else {
+      this.selection.start = 0;
+      this.selection.end = 0;
+    }
+  }
+
+  resetSelection() {
+    this.selection.start = this.target.textContent.length;
+    this.selection.end = this.target.textContent.length;
+  }
+
+  typer({ command, value /* , flags */ }) {
+    if (!this.target) { throw new Error('no target'); }
+    switch (command) {
+      case 'typing': {
```

Comment:
We can create map with methods to remove this `switch`.
We should create method for each such handler to reduce code duplication.